case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -223,6 +227,30 @@ UUID importCase(MultipartFile mpf, boolean toPublic) {
         return caseUuid;
     }
 
+    UUID createCase(UUID parentCaseUuid) {
+        try {
+            Path existingCaseFile = getCaseFile(parentCaseUuid);
+
+            UUID newCaseUuid = UUID.randomUUID();
+            Path newCaseUuidDirectory = existingCaseFile.getParent().getParent().resolve(newCaseUuid.toString());
```

Comment:
Maybe add check on existingCaseFile and existingCaseFile.getParent, and throw exception if it is null ?